sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/WindowMessagesSubscriber.java

```diff
@@ -0,0 +1,927 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LoggingEventBuilder;
+import org.reactivestreams.Subscription;
+import reactor.core.Disposable;
+import reactor.core.Disposables;
+import reactor.core.Exceptions;
+import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Operators;
+import reactor.core.publisher.SignalType;
+import reactor.core.publisher.Sinks;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicIntegerFieldUpdater;
+import java.util.concurrent.atomic.AtomicLongFieldUpdater;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import java.util.function.Consumer;
+
+/**
+ * A subscriber that split its upstream message sequence into a series of windows.
+ * <p>
+ * A window can be requested and an {@link IterableStream} instance streaming messages belongs to that window can be
+ * obtained using the API {@link #enqueueRequest(int, Duration, Sinks.Many)}.
+ * </p>
+ *
+ * @param <T> the type of messages in the window.
+ */
+final class WindowMessagesSubscriber<T> extends BaseSubscriber<T> {
```

Comment:
Since this subscriber is for type T, the naming could be generic to: WindowedSubscriber or WindowSubscriber, etc.

Reply:
The comment applies to other uses of "message" in this class because it could be Integers, etc.

Reply:
Make sense, removed it and made it more generic.